File: src/modules/audio.ts
import Base from "../base.js";
import { EventEmitter } from "events";

export default class Audio extends Base {
  constructor(start: {
    apiKey: string;
    captchaKey: string;
    options: {
      stream: boolean;
    };
  }) {
    super(start);
  }
  async music(data: {
    prompt: string;
    model?: string;
    duration?: number;
    stream?: boolean;
  }): Promise<EventEmitter | any> {
    return await this.fetch("https://api.turing.sh/audio/music", data);
  }
  async stt(data: {
    model?: string;
    audio: string;
    diarization?: boolean;
    type?: string;
    stream?: boolean;
  }): Promise<EventEmitter | any> {
    return await this.fetch("https://api.turing.sh/audio/stt", data);
  }
  async tts(data: {
    model: string;
    voice: string;
    text: string;
    language: string;
    slow?: boolean;
    stream?: boolean;
  }): Promise<EventEmitter | any> {
    return await this.fetch("https://api.turing.sh/audio/tts", data);
  }
}

File: src/modules/audio.py
import base
from pyee import EventEmitter

class Audio(Base):
    def __init__(self, start):
        super().__init__(start)

    async def music(self, data):
        response = await self.fetch('audio/music', data)
        return response

    async def stt(self, data):
        response = await self.fetch('audio/stt', data)
        return response

    async def tts(self, data):
        response = await self.fetch('audio/tts', data)
        return response
